Mage.Sets/src/mage/cards/t/TheKenrithsRoyalFuneral.java

```diff
@@ -0,0 +1,150 @@
+package mage.cards.t;
+
+import java.util.UUID;
+import mage.constants.SuperType;
+import mage.filter.common.FilterCreatureCard;
+import mage.game.Game;
+import mage.abilities.Ability;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.Cards;
+import mage.cards.CardsImpl;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.MageObject;
+import mage.abilities.SpellAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.common.cost.CostModificationEffectImpl;
+import mage.cards.Card;
+import mage.constants.*;
+import mage.game.ExileZone;
+import mage.players.Player;
+import mage.target.common.TargetCardInGraveyard;
+import mage.util.CardUtil;
+
+/**
+ *
+ * @author @stwalsh4118
+ */
+public final class TheKenrithsRoyalFuneral extends CardImpl {
+
+    public TheKenrithsRoyalFuneral(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{2}{W}{B}");
+        
+        this.addSuperType(SuperType.LEGENDARY);
+
+        // When The Kenriths' Royal Funeral enters the battlefield, exile up to two target legendary creature cards from your graveyard. You draw X cards and you lose X life, where X is the greatest mana value among cards exiled this way.
+        Ability ability = new EntersBattlefieldTriggeredAbility(new TheKenrithsRoyalFuneralEffect().setText("exile up to two target legendary creature cards from your graveyard. You draw X cards and you lose X life, where X is the greatest mana value among cards exiled this way"));
```

Comment:
you can set the text of the effect in the constructor instead of calling `setText` here